include/ruby/thread_native.h

```diff
@@ -201,5 +201,20 @@ void rb_native_cond_initialize(rb_nativethread_cond_t *cond);
  */
 void rb_native_cond_destroy(rb_nativethread_cond_t *cond);
 
+typedef struct rb_thread_hook_event_args {
+    unsigned long waiting;
+} rb_thread_hook_event_args_t;
+
+typedef void (*rb_thread_callback)(uint32_t event, rb_thread_hook_event_args_t args);
+
+typedef struct rb_thread_hook {
+    rb_thread_callback callback;
+    rb_event_flag_t event;
+
+    struct rb_thread_hook *next;
+} rb_thread_hook_t;
+
+rb_thread_hook_t * rb_thread_event_new(rb_thread_callback callback, rb_event_flag_t internal_event);
```

Comment:
could you rename them?

* `rb_thread_event_new` -> `rb_thread_internal_event_new`
* `rb_thread_event_delete` -> `rb_thread_internal_event_delete`

Reply:
Done